Solution1/BusinessLayer/Services/WeatherService.cs

```diff
@@ -54,5 +57,51 @@ public async Task<ForecastWeatherDTO> GetForecastByCityNameAsync(string cityName
 
             return _mapper.Map<ForecastWeatherDTO>(forecast).FillCommentByTemp(); 
         }
+
+        public async Task<Dictionary<bool, List<WeatherResponseDTO>>> GetWeatherByArrayCityNameAsync(string arrayCityNames)
+        {
+            var cityNames = arrayCityNames.Split(',').Select(cityName => cityName.Trim());
+            
+            var timer = new Stopwatch();
+            timer.Start();
+
+            var listTasksRequest = cityNames.Select(async cityName =>
+            {
+                var weatherResponseDTO = new WeatherResponseDTO() { CityName = cityName };
+                try
+                {
+                    ValidationCityName(cityName);
+                    weatherResponseDTO.Temp = (await _weatherApiService.GetByCityNameAsync(cityName)).TemperatureValues.Temp;
```

Comment:
what if null will be returned from service - get used to use null coalescing operator

Reply:
+